modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/CoverageNameHandling.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * GeoTools - The Open Source Java GIS Toolkit
+ * http://geotools.org
+ *
+ * (C) 2016, Open Source Geospatial Foundation (OSGeo)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation;
+ * version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ */
+package org.geotools.gce.imagemosaic;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.geotools.coverage.grid.io.GridCoverage2DReader;
+import org.geotools.gce.imagemosaic.namehandler.CoverageNameHandlerSPI;
+import org.geotools.gce.imagemosaic.namehandler.CoverageNameHandlerSpiFinder;
+
+/**
+ * Inner class delegated to parse coverageNameHandlers property and setup proper 
+ * CoverageNameHandler machinery to be used to return a target coverageName. 
+ */
+class CoverageNameHandling {
```

Comment:
I believe it would make more sense to call this handler, and the interface CoverageNameCollector